pkg/restore/crd.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RestoreCrd(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace, clusterNamespace string, args []string) {
+	crd := args[0]
+	getCrName := `kubectl -n %s get %s -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.deletionGracePeriodSeconds}{"\n"}{end}' | awk '$2=="0" {print $1}'`
+	command := fmt.Sprintf(getCrName, clusterNamespace, crd)
+	crName := strings.TrimSpace(exec.ExecuteBashCommand(command))
+
+	if len(args) == 2 {
+		crName = args[1]
+	}
+
+	logging.Info("Restoring crd %s name %s", crd, crName)
+
+	logging.Info("Scaling down the operator to 0")
+	err := debug.SetDeploymentScale(ctx, k8sclientset.Kube, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		logging.Fatal(err)
```

Comment:
> So if the operator is not running, it blocks all plugin commands? Instead, how about if we just print a warning that the version could not be detected because the operator is down. Then the commands won't be blocked. In some disaster recovery scenarios, it will likely be common that the admin could scale down the operator while they are troubleshooting, before running these commands.

Did you see this comment? thanks

Reply:
Do you want to do the above change for this command or all the controls in the plugin? I missed reading your comment earlier

Reply:
Any command that doesn't require the operator should not be blocked if the operator is not running. So I'd suggest:
1. The version check should only print a message if it couldn't detect the version. 
2. Add a new helper that waits for the operator to start, which is only called by commands that require the operator. 
3. If this call to scale down the operator fails because it's already not running, then we should continue since the intended state of stopping the operator is already in the desired state.

Reply:
And this could certainly be a separate PR, but I'd suggest we implement it before releasing this feature.